storage/txcache/maps/concurrentMap.go

```diff
@@ -148,11 +148,10 @@ func (m *ConcurrentMap) Keys() []string {
 
 	for _, chunk := range chunks {
 		chunk.mutex.RLock()
```

Comment:
I think you can move this RLock in line 148 and RUnlock from line 154 to line 156, just to avoid multiple locks/unlocks

Reply:
no, I can't because the mutex belongs to the `chunk` element from the range

Reply:
Ah, ok